src/Umbraco.Web/PublishedCache/NuCache/PublishedSnapshotService.cs

```diff
@@ -799,7 +786,7 @@ public override void Notify(DataTypeCacheRefresher.JsonPayload[] payloads)
                 }
             }
 
-            ((PublishedSnapshot)CurrentPublishedSnapshot).Resync();
+            ((PublishedSnapshot)CurrentPublishedSnapshot)?.Resync();
```

Comment:
The "?" is not here in case the cast fails - the cast better succeeds, else it would mean that we're using NuCache but somehow got a snapshot coming from another cache solution - makes no sense - it's only here to ensure that, should we use NuCache, but, in some weird situation, there is no current snapshot, we don't die.